examples/chef/common/chef-fan-control-manager.cpp

```diff
@@ -99,98 +124,213 @@ Status ChefFanControlManager::HandleStep(StepDirectionEnum aDirection, bool aWra
+    case FanControl::FanModeEnum::kOn:
+    case FanControl::FanModeEnum::kHigh: {
+        if (!kFanModeHighSpeedRange.Contains(mSpeedCurrent))
         {
-            mPercentSetting = newPercentSetting;
-
-            // Mark both the setting AND the current dirty, since the current always
-            // tracks the target for our product.
-            MatterReportingAttributeChangeCallback(mEndpoint, FanControl::Id, Attributes::PercentSetting::Id);
-            MatterReportingAttributeChangeCallback(mEndpoint, FanControl::Id, Attributes::PercentCurrent::Id);
+            DataModel::Nullable<uint8_t> speedSetting(kFanModeHighSpeedRange.High());
+            SetSpeedSetting(speedSetting);
         }
-
         break;
     }
-    default:
+    case FanControl::FanModeEnum::kSmart:
+    case FanControl::FanModeEnum::kAuto: {
+        ChipLogProgress(NotSpecified, "ChefFanControlManager::FanModeWriteCallback: Auto");
         break;
     }
-
-    // Fall through goes to attribute store legacy handling.
-    return CHIP_NO_ERROR;
+    case FanControl::FanModeEnum::kUnknownEnumValue: {
+        ChipLogProgress(NotSpecified, "ChefFanControlManager::FanModeWriteCallback: Unknown");
+        break;
+    }
+    }
 }
 
-CHIP_ERROR ChefFanControlManager::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+void ChefFanControlManager::SetSpeedSetting(DataModel::Nullable<uint8_t> aNewSpeedSetting)
 {
-    VerifyOrDie(aPath.mClusterId == FanControl::Id);
-    VerifyOrDie(aPath.mEndpointId == mEndpoint);
-
-    switch (aPath.mAttributeId)
+    if (aNewSpeedSetting.IsNull())
     {
-    case PercentCurrent::Id: {
-        // Current percents always tracks setting immediately in our implementation.
-        return aEncoder.Encode(mPercentSetting.ValueOr(0));
+        ChipLogError(NotSpecified, "ChefFanControlManager::SetSpeedSetting: invalid value");
+        return;
     }
-    case PercentSetting::Id: {
-        return aEncoder.Encode(mPercentSetting);
-    }
-    case SpeedCurrent::Id: {
-        // Current speed always tracks setting immediately in our implementation.
-        return aEncoder.Encode(mSpeedSetting.ValueOr(0));
+
+    if (aNewSpeedSetting.Value() != mSpeedCurrent)
+    {
+        Status status = FanControl::Attributes::SpeedSetting::Set(mEndpoint, aNewSpeedSetting);
+        if (status != Status::Success)
+        {
+            ChipLogError(NotSpecified, "ChefFanControlManager::SetSpeedSetting: failed to set SpeedSetting attribute: %d",
+                         to_underlying(status));
+        }
     }
-    case SpeedSetting::Id: {
-        return aEncoder.Encode(mSpeedSetting);
+}
+
+void ChefFanControlManager::Init()
+{
+    DataModel::Nullable<Percent> percentSetting = GetPercentSetting();
+    SetPercentCurrent(percentSetting.IsNull() ? 0 : percentSetting.Value());
+
+    DataModel::Nullable<uint8_t> speedSetting = GetSpeedSetting();
+    SetSpeedCurrent(speedSetting.IsNull() ? 0 : speedSetting.Value());
```

Comment:
```suggestion
    SetPercentCurrent(GetPercentSetting().ValueOr(0));
    SetSpeedCurrent(GetSpeedSetting().ValueOr(0));
```

Reply:
Thanks, Andrei. Learned lots of new tricks today!